Tests/Shared.Specs/FunctionExceptionAssertionSpecs.cs

```diff
@@ -328,6 +331,151 @@ public void When_function_throw_aggregate_in_aggregate_exception_and_that_was_no
+            // Arrange
+            //-----------------------------------------------------------------------------------------------------------
+            var watch = Stopwatch.StartNew();
+            var waitTime = 100.Milliseconds();
+            var pollInterval = 10.Milliseconds();
+
+            Func<int> throwShorterThanWaitTime = () =>
+            {
+                if (watch.Elapsed <= (waitTime.Milliseconds / 2).Milliseconds())
+                {
+                    throw new ArgumentException("An exception was forced");
+                }
+
+                return 0;
+            };
+
+            //-----------------------------------------------------------------------------------------------------------
+            // Act
+            //-----------------------------------------------------------------------------------------------------------
+            Action act = () => throwShorterThanWaitTime.Should().NotThrowAfter(waitTime, pollInterval);
+
+            //-----------------------------------------------------------------------------------------------------------
+            // Assert
+            //-----------------------------------------------------------------------------------------------------------
+            act.Should().NotThrow();
+        }
+
+        [Fact]
+        public void When_no_exception_should_be_thrown_after_wait_time_the_func_result_should_be_returned()
+        {
+            //-----------------------------------------------------------------------------------------------------------
+            // Arrange
+            //-----------------------------------------------------------------------------------------------------------
+            var watch = Stopwatch.StartNew();
+            var waitTime = 100.Milliseconds();
+            var pollInterval = 10.Milliseconds();
+
+            Func<int> throwShorterThanWaitTime = () =>
+            {
+                if (watch.Elapsed <= (waitTime.Milliseconds / 2).Milliseconds())
+                {
+                    throw new ArgumentException("An exception was forced");
+                }
+
+                return 42;
+            };
+
+            //-----------------------------------------------------------------------------------------------------------
+            // Act
+            //-----------------------------------------------------------------------------------------------------------
+            AndWhichConstraint<FunctionAssertions<int>, int> act =
+                throwShorterThanWaitTime.Should().NotThrowAfter(waitTime, pollInterval);
+
+            //-----------------------------------------------------------------------------------------------------------
+            // Assert
+            //-----------------------------------------------------------------------------------------------------------
+            act.Subject.Should().Be(42);
```

Comment:
Just to double check.
Could this be written as
```
throwShorterThanWaitTime.Should().NotThrowAfter(waitTime, pollInterval)
    .Which.Should().Be(42);
```

Reply:
Yes!